state/open.go

```diff
@@ -130,18 +128,21 @@ func newState(
 	}()
 
 	mongodb := session.DB(jujuDB)
-	sstxn := featureflag.Enabled(feature.MongoDbSSTXN)
+	sstxn := txn.SupportsServerSideTransactions(mongodb)
+
+	// TODO(wallyworld) - server side txns have issues on mongo 4.0
+	// (due to juju making bad txn.Ops, and potentially mongo 4 issues)
+	// Symptoms include intermittent success in applying a txn and reading results back,
+	// plus txn.Op slices that are wrong, eg insert dup or missing asserts.
+	// Related test failures here:
+	// https://jenkins.juju.canonical.com/job/github-make-check-juju/9317/testReport/
```

Comment:
@wallyworld, bad news.

I investigated `FAIL: deployer_test.go:187: deployerSuite.TestDeployRecallRemoveSubordinates`. I dumped the transaction ops and DB for a failure:
https://pastebin.canonical.com/p/t3f3DWY6sN/

It looks like the DB is congruent with the assertions, so unless I am mistaken there's a deeper issue than a change in txn execution leniency.